server/src/internalClusterTest/java/org/elasticsearch/search/fieldcaps/FieldCapsHasValueIT.java

```diff
@@ -0,0 +1,366 @@
+        assertThat(unmappedField, Matchers.hasKey("_index"));
+        assertEquals(
+            new FieldCapabilities("_index", "_index", true, true, true, null, null, null, Collections.emptyMap()),
+            unmappedField.get("_index")
+        );
+    }
+
+    public void testTwoFieldsNameTwoIndices() {
+        prepareIndex(INDEX1).setSource("foo", "foo-text").get();
+        prepareIndex(INDEX2).setSource("bar", 1704293160000L).get();
+        refresh(INDEX1, INDEX2);
+
+        FieldCapabilitiesResponse response = client().prepareFieldCaps().setFields("*").setIncludeFieldsWithNoValue(false).get();
+
+        assertIndices(response, INDEX1, INDEX2, INDEX3);
+        assertThat(response.get(), Matchers.hasKey("foo"));
+        assertThat(response.get(), Matchers.hasKey("bar"));
+        // Check the capabilities for the 'foo' field.
+        Map<String, FieldCapabilities> fooField = response.getField("foo");
+        assertEquals(1, fooField.size());
+        assertThat(fooField, Matchers.hasKey("text"));
+        assertEquals(
+            new FieldCapabilities("foo", "text", false, true, false, null, null, null, Collections.emptyMap()),
+            fooField.get("text")
+        );
+        // Check the capabilities for the 'bar' field.
+        Map<String, FieldCapabilities> barField = response.getField("bar");
+        assertEquals(1, barField.size());
+        assertThat(barField, Matchers.hasKey("date"));
+        assertEquals(
+            new FieldCapabilities("bar", "date", false, true, true, null, null, null, Collections.emptyMap()),
+            barField.get("date")
+        );
+    }
+
+    public void testSameFieldNameTwoIndices() {
+        prepareIndex(INDEX1).setSource("bar", "bar-text").get();
+        prepareIndex(INDEX2).setSource("bar", 1704293160000L).get();
+        refresh(INDEX1, INDEX2);
+
+        FieldCapabilitiesResponse response = client().prepareFieldCaps().setFields("*").setIncludeFieldsWithNoValue(false).get();
+
+        assertIndices(response, INDEX1, INDEX2, INDEX3);
+        assertThat(response.get(), Matchers.hasKey("bar"));
+        // Check the capabilities for the 'bar' field.
+        Map<String, FieldCapabilities> barField = response.getField("bar");
+        assertEquals(2, barField.size());
+        assertThat(barField, Matchers.hasKey("keyword"));
+        assertEquals(
+            new FieldCapabilities("bar", "keyword", false, true, true, new String[] { INDEX1 }, null, null, Collections.emptyMap()),
+            barField.get("keyword")
+        );
+        assertThat(barField, Matchers.hasKey("date"));
+        assertEquals(
+            new FieldCapabilities("bar", "date", false, true, true, new String[] { INDEX2 }, null, null, Collections.emptyMap()),
+            barField.get("date")
+        );
+    }
+
+    public void testDeletedDocsReturned() {
+        // In this current implementation we do not handle deleted documents (without a restart).
```

Comment:
I am battled about this, the test could make sense but it would work only by hardcoding a lot of variables (n. of shards, n. of data nodes) and would mean that we should make sure to create at least two segments and force a merge after deletion before restarting. I had a test that would do all of the above but was still flaky (failed 1 out of 20 runs) and decided to remove it since the "deleted documents not returned after restart" is not really a feature but more a consequence.

Reply:
Got it, thanks for explaining!